### PR TITLE
Updated phing/phing to 3.0

### PR DESCRIPTION
**Pull request by sturkel89**
I updated build.xml and composer.json to be compatible with phing/phing v3.0.

### CODE REVIEW

File: build.xml [hunk 1]
```diff
@@ -106,30 +124,39 @@
...
-    <exec dir="${srcdir}" command="${phpunit_command} -dzend.enable_gc=0" passthru="true" checkreturn="true" />
+    <exec dir="${srcdir}" executable="${phpunit_command}" passthru="true" checkreturn="true">
+      <arg line="-dzend.enable_gc=0" /> 
+    </exec>  
   </target>
 
   <!-- Set up dependencies -->
   <target name="startup" description="set up dependencies">
-    <exec command="composer install" />
+    <exec executable="composer install">
```
Comment (demiankatz) [inline build.xml]: I think "install" needs to be an arg line here.

**Review by demiankatz — CHANGES_REQUESTED** [review]
Didn't have time for a full review but noticed one small thing...

File: build.xml (hunk 1 above)
Comment (sturkel89) [reply build.xml]: Good catch! I just made that change. :)

File: build.xml [hunk 2]
```diff
@@ -40,40 +40,58 @@
...
+   <target name="phpmd">
+    <exec executable="${srcdir}/vendor/bin/phpmd">
+      <arg line="${srcdir}/src xml ${srcdir}/tests/phpmd.xml --exclude ${srcdir}/tests --reportfile ${builddir}/reports/phpmd.xml" />
+    </exec> 
   </target>
 
   <!-- PHP_Depend code analysis -->
-  <target name="pdepend">
-    <exec command="${srcdir}/vendor/bin/pdepend --jdepend-xml=${builddir}/reports/jdepend.xml --jdepend-chart=${builddir}/reports/dependencies.svg --overview-pyramid=${builddir}/reports/pdepend-pyramid.svg ${srcdir}/src" />
+   <target name="pdepend">
```
Comment (demiankatz) [inline build.xml]: This whole pdepend block is no longer used and should be removed. (Not related to your changes -- but we might as well clean it up while we're updating things).

File: build.xml [hunk 3]
```diff
@@ -106,30 +124,40 @@
...
+          <arg line="${srcdir}/vendor/bin/phpDocumentor-${phpdoc_version}.phar --cache-folder=${builddir}/docs_cache --title=&quot;VuFindHarvest API Documentation&quot; -t ${builddir}/apidocs -d ${srcdir}/src" />
+        </exec>
       </then>
     </if>
   </target>
 
   <!-- PHPUnit -->
   <target name="phpunit" description="Run tests">
-    <exec dir="${srcdir}" command="${phpunit_command} -dzend.enable_gc=0 --log-junit ${builddir}/reports/phpunit.xml --coverage-clover ${builddir}/reports/coverage/clover.xml --coverage-html ${builddir}/reports/coverage/" passthru="true" checkreturn="true" />
+    <exec dir="${srcdir}" executable="${phpunit_command}" passthru="true" checkreturn="true">
```
Comment (demiankatz) [inline build.xml]: Ere and I finished upgrading Phing to 3.0 in the main project today, and we discovered a problem: in some cases, the phpunit_command property gets set to an executable plus arguments, and that breaks the new system here. The workaround was to create a new "sh" property that specifies the Unix shell command, and then make _that_ the executable while putting everything else into the argument line.

Here's where the new property is set up at the top of the file:

https://github.com/vufind-org/vufind/blob/dev/build.xml#L3

And here is how things are put together in the phpunit commands:

https://github.com/vufind-org/vufind/blob/dev/build.xml#L369-L370

So, for example, these two lines would become:

```
<exec dir="${srcdir}" executable="${sh}" passthru="true" checkreturn="true">
  <arg line="-c &apos;${phpunit_command} -dzend.enable_gc=0 --log-junit ${builddir}/reports/phpunit.xml --coverage-clover ${builddir}/reports/coverage/clover.xml --coverage-html ${builddir}/reports/coverage/&apos;" />
```

(Note the `-c &apos;${phpunit_command} ... &apos;` wrapped around everything to make it a command to run with the shell).

**Review by demiankatz — CHANGES_REQUESTED** [review]
@sturkel89, thanks for the fixes. I've noticed a couple of new things that need attention, though you couldn't possibly have known about them (and I only learned about the second earlier today). I think once these adjustments are made, we can make this live, and the other projects should generally be pretty easy to bring in sync with this one since they're all set up quite similarly.

File: build.xml (hunk 2 above)
Comment (demiankatz) [reply build.xml]: (The pdepend tool was discontinued last year... I removed it from composer.json and our Jenkins setup, but I must have forgotten to delete it from build.xml as well).

**Review by demiankatz — APPROVED** [review]
Thanks, @sturkel89!

**Pull request merged**